src/app/_hooks/useSoundHandler.tsx

```diff
@@ -0,0 +1,185 @@
+import { useCallback, useRef, useMemo } from 'react';
+import { IUser } from '@/app/_contexts/UserTypes';
+import {
+    loadPreferencesFromLocalStorage,
+} from '@/app/_utils/ServerAndLocalStorageUtils';
+
+export type SoundAction =
+    | 'foundOpponent'
+    | 'incomingMessage'
+    | 'statefulPromptResults'
+    | 'cardClicked'
+    | 'menuButton'
+    | 'perCardMenuButton'
+    | 'yourTurn';
+
+interface SoundHandlerOptions {
+    enabled?: boolean;
+    user?: IUser | null;
+}
+
+export const useSoundHandler = (options: SoundHandlerOptions = {}) => {
+    const {
+        enabled = true,
+        user,
+    } = options;
+
+    // Store audio objects and last play time for incomingMessage only
+    const audioRefs = useRef<Map<string, HTMLAudioElement>>(new Map());
+    const lastIncomingMessageTime = useRef<number>(0);
+    // Get preferences based on user type
+    const getPreferences = () => {
+        if (user?.authenticated) {
+            return user.preferences;
+        } else {
+            // Load from localStorage for anonymous users
+            return loadPreferencesFromLocalStorage();
+        }
+    };
+
+    /**
+     * Gets the volume from localStorage (device-specific setting)
+     * @returns Volume level between 0 and 1, defaults to 0.75
+     */
+    const getVolumeFromLocalStorage = (): number => {
```

Comment:
We can condense this down a bit and get rid of the redundant else's.
```suggestion
     try {
            const storedVolume = localStorage.getItem('karabast_volume');
            if (storedVolume !== null) {
                const volume = parseFloat(storedVolume);
                // Ensure it's a valid number between 0 and 1
                if (!isNaN(volume) && volume >= 0 && volume <= 1) {
                    return volume;
                }
        } catch (error) {
            console.warn('Error reading volume from localStorage:', error);
        }
        return 0.75;
```